src/js/components/Header/Header.js

```diff
@@ -1,19 +1,62 @@
-import React from 'react';
-
+import React, { useContext, useEffect } from 'react';
+import { ThemeContext } from 'styled-components';
+import { findScrollParent, useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  useEffect(() => {
+    let scrollTop = 0;
+    const updateScrollDir = () => {
+      // target is its scroll parent
+      const target = findScrollParent(containerRef.current);
+      const container = containerRef.current;
+
+      if (target && sticky === 'scrollup') {
+        const nextScrollTop =
+          target === document ? window.pageYOffset : target.scrollTop;
+        if (scrollTop - nextScrollTop <= 0) {
+          container.style.top = `-${
```

Comment:
Think we need transition here still (and maybe position sticky? not sure on that) in order for the transition to work when user scrolls down.

Reply:
aw yeah I was trying to reset those but looks like I would have to put it back for it to work